### ISSUE TITLE
Cannot find module '@pulumi/pulumi'

### ISSUE THREAD

**Issue by Jonne** [pulumi/pulumi-provider-boilerplate#4]
Hi!

I've followed the steps in the readme to run the example, but when i run `pulumi up` I get the following exception:

```
 pulumi:pulumi:Stack (simple-test):
    error: Running program 'C:\repos\pulumi-provider-boilerplate\examples\simple' failed with an unhandled exception:
    TSError: ⨯ Unable to compile TypeScript:
    ../../sdk/nodejs/provider.ts(4,25): error TS2307: Cannot find module '@pulumi/pulumi'.

        at createTSError (C:\repos\pulumi-provider-boilerplate\examples\simple\node_modules\ts-node\src\index.ts:261:12)
        at getOutput (C:\repos\pulumi-provider-boilerplate\examples\simple\node_modules\ts-node\src\index.ts:367:40)
        at Object.compile (C:\repos\pulumi-provider-boilerplate\examples\simple\node_modules\ts-node\src\index.ts:558:11)
        at Module.m._compile (C:\repos\pulumi-provider-boilerplate\examples\simple\node_modules\ts-node\src\index.ts:439:43)
        at Module._extensions..js (internal/modules/cjs/loader.js:770:10)
        at Object.require.extensions.<computed> [as .ts] (C:\repos\pulumi-provider-boilerplate\examples\simple\node_modules\ts-node\src\index.ts:442:12)
        at Module.load (internal/modules/cjs/loader.js:628:32)
        at Function.Module._load (internal/modules/cjs/loader.js:555:12)
        at Module.require (internal/modules/cjs/loader.js:666:19)
        at require (internal/modules/cjs/helpers.js:16:16)
```

Any idea?

**Comment by leezen** [comment]
Did you have any error output when running `npm install`? What's in your `node_modules` directory?

**Comment by Jonne** [comment]
No errors, the node_modules folder looks normal and contains the @pulumi/pulumi package. I was able to make it run by copying the node_modules folder to the `sdk\nodejs` folder

**Comment by pbzdyl** [comment]
I have this issue too. Running on macOS Big Sur 11.2.3. Is there anything else I could try to check to provide more details for investigation?

**Comment by lebel-louisjacob** [comment]
> Did you have any error output when running `npm install`? What's in your `node_modules` directory?

For me, `npm i` just never halts, so I had to delete the package-lock.json file and then `npm i` halts successfully. After this, I got the error message @Jonne received (`Cannot find module '@pulumi/pulumi'`) when doing `pulumi up`, so I did what they did (i.e. copy /examples/simple/node_modules to /sdk/nodejs). However, I get the following error when trying out `pulumi up` again:

```log
Diagnostics:
  pulumi:pulumi:Stack (simple-test):
    error: Running program '/home/louisjacoblebel/src/pulumi-provider-boilerplate/examples/simple' failed with an unhandled exception:
    Error: failed to register new resource my-random [xyz:index:Random]: 2 UNKNOWN: No Major.Minor.Patch elements found
        at Object.registerResource (/home/louisjacoblebel/src/pulumi-provider-boilerplate/sdk/nodejs/node_modules/@pulumi/pulumi/runtime/resource.js:219:27)
        at new Resource (/home/louisjacoblebel/src/pulumi-provider-boilerplate/sdk/nodejs/node_modules/@pulumi/pulumi/resource.js:215:24)
        at new CustomResource (/home/louisjacoblebel/src/pulumi-provider-boilerplate/sdk/nodejs/node_modules/@pulumi/pulumi/resource.js:307:9)
        at new Random (/home/louisjacoblebel/src/pulumi-provider-boilerplate/sdk/nodejs/random.ts:32:9)
        at Object.<anonymous> (/home/louisjacoblebel/src/pulumi-provider-boilerplate/examples/simple/index.ts:4:16)
        at Module._compile (node:internal/modules/cjs/loader:1109:14)
        at Module.m._compile (/home/louisjacoblebel/src/pulumi-provider-boilerplate/examples/simple/node_modules/ts-node/src/index.ts:439:23)
        at Module._extensions..js (node:internal/modules/cjs/loader:1138:10)
        at Object.require.extensions.<computed> [as .ts] (/home/louisjacoblebel/src/pulumi-provider-boilerplate/examples/simple/node_modules/ts-node/src/index.ts:442:12)
        at Module.load (node:internal/modules/cjs/loader:989:32)
```

Navigating to /sdk/nodejs/package.json, I see this:

```json
{
    "name": "@pulumi/xyz",
    "version": "${VERSION}",
```

I don't think `${VERSION}` is a valid semantic version :thinking:. Maybe is it supposed to be replaced in the compilation process, so I can't say for sure. In any case, replacing the value with `0.0.1` seems to work for me.

**Comment by lukehoban** [comment]
Pretty sure this is due to one of the issues addressed in #15.

**Comment by Jonne** [comment]
Can confirm that the issue is indeed fixed. Thanks!
